### PR TITLE
chore: add new partitions and regions

### PR DESCRIPTION
**Pull request by TheRealAmazonKendra**
Note that the docstring in `aws-entities` states that the regions are added in the order they went live. For several, that was not accurate so I reordered them. This is unlikely to matter but I made the change so that the documentation there would be factually correct.

Not all of these regions/partitions are live yet, but they have been announced [here](https://aws.amazon.com/blogs/aws/category/regions/)

### Checklist
- [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/region-info/build-tools/generate-static-data.ts [hunk 1]
```diff
@@ -140,9 +140,11 @@ export async function main(): Promise<void> {
...
   await fs.writeFile(path.resolve(__dirname, '..', 'lib', 'built-ins.generated.ts'), lines.join('\n'));
 
-  function registerFact(region: string, name: string | string[], value: string) {
-    const factName = typeof name === 'string' ? name : `${name[0]}(${name.slice(1).map(s => JSON.stringify(s)).join(', ')})`;
-    lines.push(`    Fact.register({ region: ${JSON.stringify(region)}, name: FactName.${factName}, value: ${JSON.stringify(value)} });`);
+  function registerFact(region: string, name: string | string[], value?: string) {
+    if (value) {
+      const factName = typeof name === 'string' ? name : `${name[0]}(${name.slice(1).map(s => JSON.stringify(s)).join(', ')})`;
+      lines.push(`    Fact.register({ region: ${JSON.stringify(region)}, name: FactName.${factName}, value: ${JSON.stringify(value)} });`);
+    }
```
Comment (scanlonp) [inline packages/aws-cdk-lib/region-info/build-tools/generate-static-data.ts:147]: What's the purpose of changing this function?

**Review by comcalvi — COMMENTED** [review]
LGTM

**Review by GavinZZ — APPROVED** [review]
Let’s go!!

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Comment by TheRealAmazonKendra** [comment]
Manually merged due to mergify outage today

**Pull request merged**